src/lib/radio/radio.ts

```diff
@@ -149,6 +152,18 @@ export class MdRadioGroup implements AfterContentInit, ControlValueAccessor {
     }
   }
 
+  @Input()
+  get required(){
+    return this._required;
+  }
+
+  set required(required: boolean) {
+    this._required = required;
+
+    if (this._radios) {
+      this._radios.forEach(radio => radio.required = this.required);
+    }
```

Comment:
This is problematic if a radio button is added to the group after it has been marked as required. A better approach would be that the individual radio button uses as its `required` check:
```ts
this.required || (this.group && this.group.required)
```
([which is how `disabled` works](https://github.com/angular/material2/blob/065469ae3626bc6c99cf69100eefb30a881a7bc7/src/lib/radio/radio.ts#L350-L350))

Also, the setters need to use the `coerceBooleanProperty` function.